core/Files.cc

```diff
@@ -107,6 +107,57 @@ StrictLevel File::fileStrictSigil(string_view source) {
     }
 }
 
+pair<size_t, size_t> File::fileStrictSigilLocation(string_view source) {
+    /*
+     * StrictLevel::None: <none>
+     * StrictLevel::False: # typed: false
+     * StrictLevel::True: # typed: true
+     * StrictLevel::Strict: # typed: strict
+     * StrictLevel::String: # typed: strong
+     * StrictLevel::Autogenerated: # typed: autogenerated
+     */
```

Comment:
I split this out but wanted to keep it next to the other code which was responsible for finding the sigil. It's possible I suppose to combine them, but I tried that and it felt clunky—in particular, this is trying to find the beginning of the comment (i.e. the `#`) while the other code finds that and then walks forward to only select the string itself.

Reply:
not sure I understand. wouldn't it be possible to build one in terms of the other? or at least, factor a shared helper, and have two functions, both which post process the "parse result" of finding the comment?

Reply:
I've gone and factored them together. There are a _few_ situations in which this new approach works differently, but they're rare and weird: in particular, in situations where a user has writen, say:
```
# typed: invalid
# typed: true
```
we previously would have looked at the `typed: invalid` line, discovered that it didn't contain a sensible string, and then gone forward and found the `# typed: true` and allowed it. We now instead will only look at the first `# typed:` and treat that as canonical. I think this is an appropriate trade-off for now, and in the future I think we should additionally raise an error when `typed:` lines contain nonsensical or invalid values (which is what the TODO left in this code suggested) in which case this change actually gets us closer. But I figure it's still worth calling out as a change in semantics in some edge-cases.